cylc/flow/taskdef.py

```diff
@@ -88,9 +82,24 @@ def add_dependency(self, dependency, sequence):
                 this dependency applies.
 
         """
-        if sequence not in self.dependencies:
-            self.dependencies[sequence] = []
-        self.dependencies[sequence].append(dependency)
+        try:
+            self.dependencies[sequence].append(dependency)
+        except KeyError:
+            self.dependencies[sequence] = [dependency]
```

Comment:
I **think** this could be `self.dependencies.setdefault(sequence, []).append(dependency)`, but am not too sure (rarely use `setdefault`). Not a blocker I think.

Reply:
:+1: